hledger-lib/Hledger/Read/Common.hs

```diff
@@ -249,42 +249,29 @@ commodityStyleFromRawOpts rawOpts =
+
 -- | Given a parser to ParsedJournal, input options, file path and
 -- content: run the parser on the content, and finalise the result to
 -- get a Journal; or throw an error.
 parseAndFinaliseJournal :: ErroringJournalParser IO ParsedJournal -> InputOpts
                            -> FilePath -> Text -> ExceptT String IO Journal
-parseAndFinaliseJournal parser iopts f txt = do
-  let y = first3 . toGregorian $ _ioDay iopts
-      initJournal = nulljournal{ jparsedefaultyear = Just y, jincludefilestack = [f] }
-  eep <- liftIO $ runExceptT $ runParserT (evalStateT parser initJournal) f txt
-  -- TODO: urgh.. clean this up somehow
-  case eep of
-    Left finalParseError -> throwError $ finalErrorBundlePretty $ attachSource f txt finalParseError
-    Right ep -> case ep of
-                  Left e   -> throwError $ customErrorBundlePretty e
-                  Right pj -> journalFinalise iopts f txt pj
-
--- | Like parseAndFinaliseJournal but takes a (non-Erroring) JournalParser.
--- Also, applies command-line account aliases before finalising.
--- Used for timeclock/timedot.
--- TODO: get rid of this, use parseAndFinaliseJournal instead
-parseAndFinaliseJournal' :: JournalParser IO ParsedJournal -> InputOpts
-                           -> FilePath -> Text -> ExceptT String IO Journal
-parseAndFinaliseJournal' parser iopts f txt = do
-  let y = first3 . toGregorian $ _ioDay iopts
-      initJournal = nulljournal
-        { jparsedefaultyear = Just y
-        , jincludefilestack = [f] }
-  ep <- liftIO $ runParserT (evalStateT parser initJournal) f txt
-  -- see notes above
-  case ep of
-    Left e   -> throwError $ customErrorBundlePretty e
-    Right pj ->
-      -- apply any command line account aliases. Can fail with a bad replacement pattern.
-      case journalApplyAliases (aliasesFromOpts iopts) pj of
-        Left e    -> throwError e
-        Right pj' -> journalFinalise iopts f txt pj'
+parseAndFinaliseJournal parser iopts f txt =
+    initialiseAndParseJournal parser iopts f txt >>= journalFinalise iopts f txt
+
+-- | Given a parser to ParsedJournal, input options, file path and
+-- content: run the parser on the content. This is all steps of
+-- 'parseAndFinaliseJournal' without the finalisation step, and is used when
+-- you need to perform other actions before finalisation, as in parsing
+-- Timeclock and Timedot files.
+initialiseAndParseJournal :: ErroringJournalParser IO ParsedJournal -> InputOpts
+                          -> FilePath -> Text -> ExceptT String IO Journal
+initialiseAndParseJournal parser iopts f txt =
+    prettyParseErrors $ runParserT (evalStateT parser initJournal) f txt
+  where
+    y = first3 . toGregorian $ _ioDay iopts
+    initJournal = nulljournal{jparsedefaultyear = Just y, jincludefilestack = [f]}
+    prettyParseErrors = withExceptT customErrorBundlePretty . liftEither
+                    <=< withExceptT (finalErrorBundlePretty . attachSource f txt)
```

Comment:
Woah, prettyParseErrors.. what is going on there, perhaps a short comment for future devs ?

Reply:
What do you mean? It's completely self-explanatory. (end sarcasm). Okay, comment and type signature added.